plugins/out_pgsql/pgsql_connections.c

```diff
@@ -73,6 +73,19 @@ void *pgsql_create_connection(struct flb_pgsql_config *ctx)
     flb_plg_info(ctx->ins, "switching postgresql connection "
                  "to non-blocking mode");
 
+    if (ctx->schema !== FLB_PGSQL_SCHEMA) {
```

Comment:
I would probably make sure that we're using the schema every time, not just when we know it was changed, otherwise we shouldn't set the schema and the default should be blank, meaning, the default schema in the PostgreSQL configuration

Reply:
Are you saying that we should change it this way?

```
res = PQexec(conn->conn, "SET search_path = %s", ctx->schema);
if (PQresultStatus(res) != PGRES_COMMAND_OK)
    {
        flb_plg_error(ctx->ins,
              "failed set search_path=%s with error: %s",
              ctx->schema, PQerrorMessage(conn->conn));
        PQfinish(conn->conn);
        flb_free(conn);
        return NULL;
    }
```

When the default value (blank) is set to search_path each time, the schema value will be blank.

```
# SELECT current_schema();
 current_schema
----------------
 public
(1 row)

# set search_path = '';
SET
# SELECT current_schema();
 current_schema
----------------
(1 row)
```

Reply:
Note you've also got a typo in that comparison above: `!==`